monkey/monkey_island/cc/repository/utils.py

```diff
@@ -33,3 +35,34 @@ def initialize_machine_repository(machine_repository: IMachineRepository):
             hostname=gethostname(),
         )
         machine_repository.upsert_machine(machine)
+
+
+DOT_REPLACEMENT = ",,,"
+
+
+def mongo_dot_encoder(mapping: Mapping[str, Any]) -> Mapping[str, Any]:
+    """
+    Mongo can't store keys with "." symbols (like IP's and filenames). This method
+    replaces all occurances of "." with ",,,"
+    :param mapping: Mapping to be converted to mongo compatible mapping
+    :return: Mongo compatible mapping
+    """
+    mapping_json = json.dumps(mapping)
+    if DOT_REPLACEMENT in mapping_json:
+        raise StorageError(
+            f"Mapping {mapping} already contains {DOT_REPLACEMENT}."
+            f" Aborting the encoding procedure"
+        )
```

Comment:
Is this necessary? The next line will just move on silently if it finds no dots.

Reply:
If I'm missing something and it is necessary, update the docstring to mention that this error could be raised.

Reply:
Raising an error is necessary, because we can't encode and decode a string like "these are three commas: ,,,". Originally the string was "these are three commas: ,,,", but on decode it will be "these are three commas: .". Whatever we use as a replacement for `.` can't be in the original string or we lose data